### PR TITLE
Add python 3 enum handling

### PR DESCRIPTION
**Pull request by Anna-Livia**
Connected to openfisca/openfisca-france#831

* Technical improvement. 
* Impacted areas: `openfisca_country_template/variables/housing.py`, openfisca_country_template/variables/taxes.py
* Details:
  - EnumCols (e.g. housing_occupancy_status) , now reference an Enum Class (e.g. HousingOccupancyStatus) that has a name & description for each item and an explicit default value.

These changes:

- Impact the OpenFisca-Country-Template public API (for instance renaming or removing a variable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,43 @@
 # Changelog
 
+## 2.0.0 - [#24](https://github.com/openfisca/country-template/pull/24)
+
+* Technical change
+* Details:
+  - Introduce the use of a keywords to reference enum item.
+  - EnumCols (e.g. housing_occupancy_status), now reference an Enum Class (e.g. HousingOccupancyStatus) that has a name & description for each item and an explicit default value.
+
+Remove:
```
Comment (fpagnoux) [inline CHANGELOG.md]: The Changelog of a package is mainly targeted to _re-users_. When reading a block as a user, I should get answers to the questions : 

- Am I impacted by this change ? 
- What new feature can I use ?
- What feature I used to use is now deprecated / remove ? What should I use instead ? 
- How has the _public API_ of the package has changed ? Did the format of the input / output change ?

From the country-template's user perspective, the fact that the syntax of enums changed is not significant. It matters for the _maintener_ of this package, but the _public API_  is not impacted.

However, the _public API_ is impacted by the fact that, when setting an input enum variable, I am now supposed to set `free_lodger`, and can't use `2` or `Free lodger` anymore.

The same goes for the output : when requesting the calculation of an enum variable, I should now be expecting `free_lodger`, and not `Free lodger`.

These two changes impact me if I use the Web API, or the python API, or when I write YAML tests. These are the _real breaking changes_ that need to be documented in this changelog 🙂 .

**Comment by Anna-Livia** [comment]
Rebased with V20

File: openfisca_country_template/tests/housing_tax.yaml [hunk 2]
```diff
@@ -1,18 +1,32 @@
 # Test files describe situations and their expected outcomes
 # We can run this test on our command line using `openfisca-run-test housing_tax.yaml`
 
-- name: Household living in a 40 sq.meters accomodation
+- name: Household living in a 40 sq.meters accomodation while tenants
```
Comment (sandcha) [inline openfisca_country_template/tests/housing_tax.yaml]: `Tenants living in`...?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,77 @@
 # Changelog
 
+## 2.0.0 - [#24](https://github.com/openfisca/country-template/pull/24)
+
+* Technical change
```
Comment (sandcha) [inline CHANGELOG.md]: `Breaking change`?
Besides, on the user side, `HousingOccupancyStatus` type seems to be the main change.
So, what about adding a reference to the documentation on enumerations and remove generic information on enumerations in the changelog?
> Remove details but keep web api, yaml and python examples?
> For python example, use `housing_occupancy_status` and remove text on `name` and `value`?

Comment (Anna-Livia) [reply CHANGELOG.md]: Users of the API will not see HousingOccupancyStatus.
The breaking change is, IMHO, in the name/value paradigm.

File: openfisca_country_template/tests/housing_tax.yaml [hunk 4]
```diff
@@ -1,18 +1,32 @@
 # Test files describe situations and their expected outcomes
 # We can run this test on our command line using `openfisca-run-test housing_tax.yaml`
 
-- name: Household living in a 40 sq.meters accomodation
+- name: Tenantd living in a 40 sq.meters accomodation
```
Comment (sandcha) [inline openfisca_country_template/tests/housing_tax.yaml]: Typo on `Tenant`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,79 @@
 # Changelog
 
+## 2.0.0 - [#24](https://github.com/openfisca/country-template/pull/24)
+
+#### breaking change
+
+* Details:
+  - Upgrade to Core v21
+  - Introduce the use of a string identifier to reference enum item.
+  - When setting an Enums (e.g. housing_occupancy_status), set the relevant string identifier (e.g. `free_lodger`). Indexes (e.g.`2`) and phrases (e.g. `Free Lodgers`) cannot be used anymore.
```
Comment (sandcha) [inline CHANGELOG.md]: > Start by the broken feature then explain what to do to adapt?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,79 @@
...
+        housing_occupancy_status: "free_lodger"
+    }
+}
+```
+
+#### YAML testing
+Before:
+
+```
+name: Household living in a 40 sq.meters accomodation while free_lodgers
```
Comment (sandcha) [inline CHANGELOG.md]: `Free Lodger`? Did `free_lodgers` exist before?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,79 @@
...
+  period: 2017
+  input_variables:
+    accomodation_size:
+      2017-01: 40
+    housing_occupancy_status:
```
Comment (sandcha) [inline CHANGELOG.md]: Quotes around `housing_occupancy_status`?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,79 @@
...
+      2017-01: free_lodger
+  output_variables:
+    housing_tax: 0
+```
+
+#### Python API
+
+When calculating an enum variable in Python, the output will be an array of enum items.
```
Comment (sandcha) [inline CHANGELOG.md]: > `Enum` items? (Enum type)

File: CHANGELOG.md (hunk 7 above)
Comment (Anna-Livia) [reply CHANGELOG.md]: Not in the Yaml :)

File: CHANGELOG.md (hunk 5 above)
Comment (Anna-Livia) [reply CHANGELOG.md]: I like starting with the new feature :)

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,80 @@
...
+Now:
+
+```
+"persons": {
+    "Bill": {}
+},
+"households": {
+    "_": {
+        "parent": ["Bill"]
+        housing_occupancy_status: "free_lodger"
```
Comment (sandcha) [inline CHANGELOG.md]: No quotes here either? 😳

**Pull request merged**